sbt-scalafmt/src/main/scala/com/lucidchart/sbt/scalafmt/ScalafmtCorePlugin.scala

```diff
@@ -163,15 +170,22 @@ object ScalafmtCorePlugin extends AutoPlugin {
 
             val scalafmtter = scalafmtFn.value
             val failForStyleIssues = scalafmtFailTest.value
-            val differentCount = sources.value.count { file =>
-              val content = IO.read(file)
-              val hasChanges = content != scalafmtter(file.toString, content)
-              if (hasChanges) {
-                val msg = s"$file has changes after scalafmt"
-                if (failForStyleIssues) logger.error(msg)
-                else logger.warn(msg)
-              }
-              hasChanges
+            val showDiff = scalafmtShowDiff.value
+            val differentCount = sources.value.count {
+              file =>
+                val original = IO.read(file)
+                val formatted = scalafmtter(file.toString, original)
+                val hasChanges = original != formatted
+                if (hasChanges) {
+                  val msg = if (showDiff) {
+                    val diff = ColorDiff(original.split('\n').toList, formatted.split('\n').toList)
+                    s"$file has changes after scalafmt:\n$diff"
+                  } else s"$file has changes after scalafmt"
+                  logger.warn(msg)
```

Comment:
Is this change going to logger.warn(msg) twice if failForStyleIssues is false?

It looks like if we remove line 184, then we'll have the logging behavior we used to have. Is that right, or am I missing something?

Reply:
thanks for catching that, you're right. I just removed that line.